shared/js/devtools/panel.es6.js

```diff
@@ -6,10 +6,46 @@ const tabPicker = document.getElementById('tab-picker')
 const tdsOption = document.getElementById('tds')
 const displayFilters = document.querySelectorAll('#table-filter input')
 
+/**
+ * @param {Node} element
+ * @returns {HTMLTableRowElement}
+ */
+function assertTableRowElement (element) {
+    // @ts-ignore
+    return element
+}
+
 function sendMessage (messageType, options, callback) {
     chrome.runtime.sendMessage({ messageType, options }, callback)
 }
 
+/**
+ * @param {(m: any) => HTMLTableRowElement} f
+ * @returns {(m: any) => void}
+ */
+function addRequestRow (f) {
+    return (m) => {
+        const row = f(m)
+        if (row) {
+            // if duplicate request lines would be printed, we instead show a counter increment
+            const prevRow = document.querySelector('tbody > tr:last-child')
+            if (prevRow) {
+                const prevRowCopy = assertTableRowElement(prevRow.cloneNode(true))
+                prevRowCopy.querySelector('.action-count').textContent = ''
```

Comment:
This handling is a little fragile if the table structure changes this will stop working.

So either:

- Add tests
- Run the table rendering from a data storage that can be interrogated for the previous row.

I think as we're only rolling up by the previous row; simply store row data in a _lastRow_ object and compare their fields.

Reply:
@jonathanKingston I've added some helpers, w/ the row linked as an object